mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -189,11 +232,21 @@ class ChiSqSelector @Since("1.3.0") (
    */
   @Since("1.3.0")
   def fit(data: RDD[LabeledPoint]): ChiSqSelectorModel = {
-    val indices = Statistics.chiSqTest(data)
-      .zipWithIndex.sortBy { case (res, _) => -res.statistic }
-      .take(numTopFeatures)
-      .map { case (_, indices) => indices }
-      .sorted
+    val chiSqTestResult = Statistics.chiSqTest(data)
+    val features = selectorType match {
+      case ChiSqSelectorType.KBest => chiSqTestResult
+        .zipWithIndex.sortBy { case (res, _) => -res.statistic }
```

Comment:
Hi @srowen , do you mean we should move the sort from the model to here?
There are two sort here. 
One is sort the indices, which we have been moved from here to the model. 
The second is sort the features, because the KBest, and Percentile is selection by the order of  res.statistic, so we sort the features here.  FPR is selection by res.p-value > alpha, so don't need sort here.